commands/operator-sdk/cmd/test/local.go

```diff
@@ -151,7 +175,7 @@ func testLocalFunc(cmd *cobra.Command, args []string) {
 	if tlConfig.goTestFlags != "" {
 		testArgs = append(testArgs, strings.Split(tlConfig.goTestFlags, " ")...)
 	}
-	if tlConfig.namespace != "" {
+	if tlConfig.namespace != "" || tlConfig.noSetup {
```

Comment:
What if the `namespace` flag is not set and `no-setup=true`.
I don't think we check that if `no-setup==true` then `namespace` must be set above.

Reply:
I got a bit confused. The namespace flag is not required for the `no-setup` flag, because the test-framework will use the default namespace specified by the kubeconfig if the singleNamespace flag is enabled but the env var is empty.

Reply:
If users are explicitly skipping the operator setup with `--no-setup=true` then they'll almost always be pointing to the namespace where the operator is setup with `--namespace`.

I think we can explicitly couple `--no-setup` to require a target `--namespace`, rather than having the test-framework use the default namespace found in the kubeconfig which is implicit.

Reply:
After offline discussions it's fine to keep them decoupled so that users can use the default namespace in their kubeconfig if needed.